Bobcat/Profile.cpp

```diff
@@ -350,6 +352,42 @@ void Profiles::Add()
 	Sync();
 }
 
+void Profiles::Clone()
+{
+	if(String s; EditTextNotNull(s, t_("Clone Profile"), t_("Name"))) {
+		if(!list.FindSetCursor(s)) {
+			Profile source = list.Get(1).To<Profile>();
+			Profile p(source);
+			p.name = s;
+			auto& m = GetHyperlinkPatterns();
+			int i = m.FindAdd(p.name);
+			m[i] <<= m.Get(source.name);
+			list.Add(s, RawToValue(p));
+			list.GoEnd();
+		}
+	}
+	Sync();
+}
+
+void Profiles::Rename()
+{
+	if(String s(list.Get(0)); EditTextNotNull(s, t_("Rename Profile"), t_("New name"))) {
+		if(!list.FindSetCursor(s)) {
+			Profile p = list.Get(1).To<Profile>();
+			auto& m = GetHyperlinkPatterns();
+			String f = ProfileFile(p.name);
+			if(FileExists(f))
+				DeleteFile(f);

```

Comment:
This is wrong from the UX point of view. The file should not be deleted, until the user closes th Settings dialog with OK button. Because if he changes his mind a clicks on Cancel, he'll be quite surprised that the profile disappeared.

And the same applies at least partially for the profile removal. In that case there is at least the confirmation dialog, but it would be nicer and more consistent if the irreversible changes were made only when the Settings are applied by the OK button.

I'll try to address this soon too.

Reply:
Yep, you are right.